compiler/rustc_trait_selection/src/solve/trait_goals.rs

```diff
@@ -168,6 +168,24 @@ impl<'tcx> assembly::GoalKind<'tcx> for TraitPredicate<'tcx> {
         }
     }
 
+    fn consider_builtin_fn_ptr_trait_candidate(
+        ecx: &mut EvalCtxt<'_, 'tcx>,
+        goal: Goal<'tcx, Self>,
+    ) -> QueryResult<'tcx> {
+        if goal.predicate.self_ty().has_non_region_infer() {
+            return ecx.make_canonical_response(Certainty::AMBIGUOUS);
+        }
```

Comment:
Why? `fn(_): FnPtr` should hold unambiguously?

Reply:
If this is just to handle the `_: FnPtr` case, then that's already handled elsewhere in the solver.

Reply:
Oops copy paste bug.

Reply:
Same for erasing regions a couple of lines below